src/librustc/mir/interpret/error.rs

```diff
@@ -139,9 +139,11 @@ impl<'tcx> ConstEvalErr<'tcx> {
     ) -> Result<DiagnosticBuilder<'tcx>, ErrorHandled> {
         match self.error {
             InterpError::Layout(LayoutError::Unknown(_)) |
-            InterpError::TooGeneric => return Err(ErrorHandled::TooGeneric),
+            InterpError::InvalidProgram(InvalidProgramMessage::TooGeneric) =>
```

Comment:
I suggest to `use InvalidProgramMessage::*` before this match, to keep it a bit shorter.